lib/fluent/plugin/out_forward.rb

```diff
@@ -880,6 +1063,32 @@ def on_read(sock, ri, data)
           raise "BUG: unknown session state: #{ri.state}"
         end
       end
+
+      private
+
+      def connect(host = nil)
+        sock = if @keepalive
+                 @socket_cache.fetch_or { @sender.create_transfer_socket(host || resolved_host, port, @hostname) }
+               else
+                 @log.debug('connect new socket')
+                 @sender.create_transfer_socket(resolved_host, port, @hostname)
```

Comment:
Is this point also `host || resolved_host`?

Reply:
Right https://github.com/fluent/fluentd/pull/2393/commits/44cb30170888223fb452ac5858780c95bcff8858